tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -50,6 +62,18 @@ static void is_usermode(void)
 	zassert_true(_is_user_context(), "thread left in kernel mode\n");
 }
 
+void entry(void)
+{
+	zassert_unreachable(" Moving to user mode from user mode"
+			" did not fault\n");
+}
+
+static void goto_usermode(void)
+{
+	/* Try to go usermode from usermode */
```

Comment:
I think it's worth noting that entering user mode, when you are already in user mode, should not have specific runtime checks for it. Some platforms may crash, it's really undefined behavior. I'd like to see this removed and at most, checked in the kernel with an _ASSERT